### PR TITLE
Feature/disassembler

### PR DESCRIPTION
**Pull request by zephyyrr**
Disassembler for soda-code to aspartasm.

### CODE REVIEW

File: aspartasm/cola/main.go [hunk 1]
```diff
@@ -35,13 +47,34 @@ func main() {
...
+			log.Println("Mode:", "Disassemble-only")
+		}
+
+		ins, err := aspartasm.ReadInstructions(in)
+
+		if err != nil {
+			log.Fatalf("Error disassembling %s: %s", flag.Arg(0), err)
+		}
+
+		w := bufio.NewWriter(out)
```
Comment (zephyyrr) [inline aspartasm/cola/main.go]: Any particular reason to using bufio for this?
I would have used the fmt package and fmt.Fprintln(out, i)

**Comment by zephyyrr** [comment]
I added the new instructions added since this branch started

File: aspartasm/ins.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+// A register
+type Reg byte
+
+func (r Reg) String() string {
+	return fmt.Sprintf("r%d", r)
+}
+
+// An immediate value
+type Imm uint32
```
Comment (zephyyrr) [inline aspartasm/ins.go]: Why is a Immediate value a uint32 when a immediate is really a int16?

File: aspartasm/operations.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+}
+
+func oneRegImm(in *bufio.Reader) ([]Arg, error) {
+	r1, err := in.ReadByte()
+
+	if err != nil {
+		return nil, err
+	}
+
+	var imm uint16
```
Comment (zephyyrr) [inline aspartasm/operations.go]: Should be "var imm Imm"

File: aspartasm/operations.go [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+}
+
+func oneImm(in *bufio.Reader) ([]Arg, error) {
+	_, err := in.ReadByte()
+
+	if err != nil {
+		return nil, err
+	}
+
+	var imm uint16
```
Comment (zephyyrr) [inline aspartasm/operations.go]: See line 339

File: aspartasm/cola/main.go (hunk 1 above)
Comment (Victorystick) [inline aspartasm/cola/main.go]: I can change that.

**Comment by zephyyrr** [comment]
This fix #12.

**Pull request merged**